keel-core/src/test/kotlin/com/netflix/spinnaker/keel/actuation/NewEnvironmentPromotionCheckerTests.kt

```diff
@@ -0,0 +1,321 @@
+package com.netflix.spinnaker.keel.actuation
+
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.artifacts.DockerArtifact
+import com.netflix.spinnaker.keel.api.artifacts.TagVersionStrategy.SEMVER_TAG
+import com.netflix.spinnaker.keel.core.api.DependsOnConstraint
+import com.netflix.spinnaker.keel.core.api.PinnedEnvironment
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.telemetry.ArtifactVersionApproved
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.runBlocking
+import org.springframework.context.ApplicationEventPublisher
+import strikt.api.expectCatching
+import strikt.assertions.isSuccess
+
```

Comment:
Will rename to just `EnvironmentPromotionCheckerTests` once I remove the old ones (separate PR)